src/test/java/hw2/ex1/Exercise1.java

```diff
@@ -0,0 +1,118 @@
+        for (WebElement element : headerItems) {
+            actualHeaderItemsTexts.add(element.getText());
+        }
+
+        for (WebElement element : headerItems) {
+            sa.assertTrue(element.isDisplayed());
+        }
+        sa.assertEquals(actualHeaderItemsTexts, expectedHeaderItemsTexts);
+
+        // 6. Assert that there are 4 images on the Index Page and they are displayed
+
+        List<WebElement> images = driver.findElements(By.cssSelector(".benefit-icon"));
+
+        for (WebElement element : images) {
+            sa.assertTrue(element.isDisplayed());
+        }
+
+        // 7. Assert that there are 4 texts on the Index Page under icons and they have proper text
+
+        List<WebElement> iconTexts = driver.findElements(By.className("benefit-txt"));
+        List<String> expectedIconTexts = Arrays.asList("To include good practices\n" +
+                        "and ideas from successful\n" +
+                        "EPAM project",
+                "To be flexible and\n" +
+                        "customizable", "To be multiplatform",
+                "Already have good base\n"+
+                "(about 20 internal and\n" +
+                 "some external projects),\n" +
+                  "wish to get more…");
+        List<String> actualIconTexts = new ArrayList<>();
+
+        for (WebElement element : iconTexts) {
+            actualIconTexts.add(element.getText());
+        }
+
+        for (WebElement element : iconTexts) {
+            sa.assertTrue(element.isDisplayed());
+        }
+
+        sa.assertEquals(actualIconTexts, expectedIconTexts);
+
+        // 8, 9, 10 steps
+
+        WebElement iframe = driver.findElement(By.id("frame"));
+
+        // 8. Assert that there is the iframe with “Frame Button” exist
+        sa.assertTrue(iframe.isDisplayed());
+
+        driver.switchTo().frame(iframe);
+        WebElement frameButton = driver.findElement(By.id("frame-button"));
+
+        // 9. Switch to the iframe and check that there is “Frame Button” in the iframe
+        sa.assertTrue(frameButton.isDisplayed());
+
+        // 10. Switch to original window back
+        driver.switchTo().defaultContent();
+
+        // 11. Assert that there are 5 items in the Left Section are displayed and they have proper text
+
+        List<WebElement> leftSectionItems = driver.findElements(By.cssSelector(".sidebar-menu"));
+        List<String> actualLeftSectionItems = Arrays.asList("Home", "Contact form", "Service",
+                "Metals & Colors", "Elements packs");
+        List<String> expectedLeftSectionItems = new ArrayList<>();
+        for (WebElement element : leftSectionItems) {
+            expectedLeftSectionItems.add(element.getText());
+        }
+
+        for (WebElement element : leftSectionItems) {
+            sa.assertTrue(element.isDisplayed());
+        }
+
+        sa.assertEquals(actualLeftSectionItems, expectedLeftSectionItems);
+
```

Comment:
How do we check that all our soft assertion work properly?

Reply:
"sa.assertAll();" is added